### PR TITLE
Add half-step angular velocity calculation

### PR DESCRIPTION
**Pull request by shahabgol**
# Description of the problem

- Previously, in the half-step calculations of the velocity-verlet integration method, the half-step angular velocity wasn't calculated. In this PR we add the calculation of the half-step angular velocity to velocity-verlet integration method.

### CODE REVIEW

File: include/dem/explicit_euler_integrator.h [hunk 1]
```diff
@@ -54,15 +54,19 @@ class ExplicitEulerIntegrator : public Integrator<dim>
...
+   * @param force Force acting on particles
+   * @param MOI A container of moment of inertia of particles
    */
   virtual void
   integrate_half_step_location(
     Particles::ParticleHandler<dim> &particle_handler,
     Tensor<1, dim> &                 body_force,
+    double                           time_step,
+    std::vector<Tensor<1, dim>> &    momentum,
     std::vector<Tensor<1, dim>> &    force,
```
Comment (blaisb) [inline include/dem/explicit_euler_integrator.h:68]: Make arguments that should be const const :)

File: include/dem/explicit_euler_integrator.h [hunk 2]
```diff
@@ -54,15 +54,19 @@ class ExplicitEulerIntegrator : public Integrator<dim>
    * @param particle_handler The particle handler whose particle motion we wish
    * to integrate
    * @param body_force A constant volumetric body force applied to all particles
-   * @param force Force acting on particles
    * @param time_step The value of the time step used for the integration
+   * @param momentum Momentum of particles
+   * @param force Force acting on particles
+   * @param MOI A container of moment of inertia of particles
    */
   virtual void
```
Comment (blaisb) [inline include/dem/explicit_euler_integrator.h:62]: Should this function be virtual?

**Review by blaisb — COMMENTED** [review]
Good job. Just take the opportunity to make the const argument const :) then it is good to go

**Review by tonielgeitani — APPROVED** [review]

File: include/dem/explicit_euler_integrator.h (hunk 1 above)
Comment (shahabgol) [reply include/dem/explicit_euler_integrator.h]: done.

File: include/dem/explicit_euler_integrator.h (hunk 2 above)
Comment (shahabgol) [reply include/dem/explicit_euler_integrator.h:62]: yes, because it is rewritten by different integration methods.

**Comment by shahabgol** [comment]
The const parameters are defined in the integrators. :)

**Review by blaisb — APPROVED** [review]

**Pull request merged**